components/elm/src/biogeochem/AllocationMod.F90

```diff
@@ -1070,24 +865,16 @@ end subroutine Allocation1_PlantNPDemand
  subroutine Allocation2_ResolveNPLimit (bounds, num_soilc, filter_soilc  , &
                             num_soilp, filter_soilp                         , &
                             cnstate_vars                                    , &
-                            carbonstate_vars, carbonflux_vars               , &
-                            nitrogenstate_vars, nitrogenflux_vars           , &
-                            phosphorusstate_vars,phosphorusflux_vars        , &
-                            soilstate_vars,waterstate_vars, &
-                            elm_fates)
```

Comment:
@peterdschwartz I added this argument (elm_fates) in PR #3990.  We don't use this argument now, but in my next set of changes I will need it.  Although I suppose I can just add it back in again, I will probably work on top of this PR or its merged result depending on timing of things.

Reply:
The way I have been handling elm_fates arguments is to place a 'use elm_instMod , only : alm_fates'   at the top of those modules.   This makes it easy to wrap any fates subroutine calls in  #ifndef _OPENACC flags.   

Do you think this is an acceptable solution or does it lose needed functionality for fates?  if so, then I can revert those changes back to passing alm_fates as an argument.

Reply:
ok, this should work. We will just need to pass in the thread index then.